### PR TITLE
Compat Sog - Fix compatibility with grad trenches

### PR DESCRIPTION
**Pull request by PabstMirror**
Grad trenches does
`delete ace_trenches_digEnvelopeSmall;`

which fails because sog compat was extending the class
`Cannot delete class ace_trenches_digEnvelopeSmall, it is referenced somewhere (used as a base class probably).`

This PR fixes compatibly by only eliminating the inheritance and also only showing the trench actions if the base action still exists.

### CODE REVIEW

File: optionals/compat_sog/XEH_postInit.sqf [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 [QGVAR(handlePunjiTrapDamage), LINKFUNC(handlePunjiTrapDamage)] call CBA_fnc_addEventHandler;
 
+GVAR(showTrenchActions) = isClass (configFile >> "CfgVehicles" >> "CAManBase" >> "ACE_SelfActions" >> "ACE_Equipment" >> "ace_trenches_digEnvelopeSmall");
```
[the remark this answers is not in the document]
Comment (PabstMirror) [reply optionals/compat_sog/XEH_postInit.sqf:5]:
```suggestion
GVAR(showTrenchActions) = isClass (configFile >> "CfgVehicles" >> "CAManBase" >> "ACE_SelfActions" >> "ACE_Equipment" >> QEGVAR(trenches,digEnvelopeSmall));
```

**Comment by Drofseh** [comment]
If Grad deleting an Ace class not something that Grad should be fixing?
Not sure why this is our problem

**Comment by PabstMirror** [comment]
grad would need to make a compat pbo just for ace's sog compat

**Comment by veteran29** [comment]
You can delete non existent classes without any error.

**Comment by PabstMirror** [comment]
This leads to a mod load order problem. I tested grad doing
```
                delete ace_compat_sog_digSpiderhole;
                delete ace_compat_digSpiderholeAngled;
                delete ace_compat_digSpiderholeDual;
                delete ace_trenches_digEnvelopeSmall;
                delete ace_trenches_digEnvelopeBig;
```
but because it doesn't require compat_sog, it may get loaded loaded first, do nothing
then compat_sog runs and add the classes:

**Comment by PabstMirror** [comment]
also restore some macros used by grad that were removed

**Pull request closed without merging**